test/jit/test_list_dict.py

```diff
@@ -1288,6 +1288,14 @@ def update(a, b):
         self.checkScript(update, (self.dict(), self.dict()))
         self.checkScript(update, (self.dict(), self.dict2()))
 
+        def upd() -> Dict[str, int]:
```

Comment:
Nit, make it into its own testcase for better parallel execution/clearer error message etc.